storage/api/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteStorageManager.java

```diff
@@ -121,14 +121,17 @@ InputStream fetchLogSegment(RemoteLogSegmentMetadata remoteLogSegmentMetadata,
 
     /**
      * Returns the index for the respective log segment of {@link RemoteLogSegmentMetadata}.
+     * <p>
+     * Note: The transaction index may not exist because segments created prior to version 2.8.0 or non-transactional topics...etc.
+     * In this case, it should still return an InputStream with empty content, instead of returning {@code null}.
```

Comment:
Encountered an error with a RSM returning `null`, since the indexCache will assert not null index.

Reply:
As the javadoc says it needs to return an empty input-stream.

Reply:
Not clear on this line:

>The transaction index may not exist because segments created prior to version 2.8.0 

Only the producer snapshots are not expected prior v2.8.0 not the transaction index. Correct me if I'm wrong.

Reply:
Changed to `The transaction index may not exist because of non-transactional topics.`. Thanks.

Reply:
Right, transaction index can be null, but there is an existing https://issues.apache.org/jira/browse/KAFKA-14993 to address that.

Reply:
Added a note in [KAFKA-14993](https://issues.apache.org/jira/browse/KAFKA-14993): 

`Note, please remember to update the javadoc in RSM after the fix. See: https://github.com/apache/kafka/pull/14352`

Thanks for pointing to the ticket.